File: src/CSharp/801-900/896.MonotonicArray.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.MonotonicArray
{
public class Solution {
    public bool IsMonotonic(int[] A) {
         return increasing(A) || decreasing(A);
    }

    public bool increasing(int[] A) {
        for (int i = 0; i < A.Length - 1; ++i)
            if (A[i] > A[i+1]) return false;
        return true;
    }

    public bool decreasing(int[] A) {
        for (int i = 0; i < A.Length - 1; ++i)
            if (A[i] < A[i+1]) return false;
        return true;
    }
}
}

File: src/Python/801-900/896.MonotonicArray.py

class Solution:
    def isMonotonic(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        flag = []
        for i in range(len(A) - 1):
            if A[i + 1] != A[i]:
                flag.append(A[i + 1] > A[i])
        for i in range(len(flag) - 1):
            if flag[i] != flag[i + 1]:
                return False
        return True


if __name__ == '__main__':
    solution = Solution()
    print(solution.isMonotonic([1,2,2,3]))
    print(solution.isMonotonic([6,5,4,4]))
    print(solution.isMonotonic([1,3,2]))
    print(solution.isMonotonic([1,2,4,5]))
else:
    pass
